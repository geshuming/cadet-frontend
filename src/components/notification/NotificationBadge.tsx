import { Intent, Tag } from '@blueprintjs/core';
import * as React from 'react';

type OwnProps = {
  large?: boolean;
  number: number;
};

const NotificationBadge: React.SFC<OwnProps> = props => (
  <Tag intent={Intent.DANGER} round={true} large={props.large}>
    {props.number}
  </Tag>
);

export default NotificationBadge;
