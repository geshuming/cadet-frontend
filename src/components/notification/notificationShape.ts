import { AssessmentCategory } from '../assessment/assessmentShape';

export type Notification = {
  assessment_id?: number;
  assessment_type?: AssessmentCategory;
  assesssment_title?: string;
  id: number;
  question_id?: number;
  submission_id?: number;
  type: NotificationType;
};

enum NotificationTypes {
  new = 'new',
  deadline = 'deadline',
  autograded = 'autograded',
  graded = 'graded',
  submitted = 'submitted',
  unsubmitted = 'unsubmitted',
  new_message = 'new_message'
}

export type NotificationType = keyof typeof NotificationTypes;
