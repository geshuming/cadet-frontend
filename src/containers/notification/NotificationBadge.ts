import { connect, Dispatch, MapDispatchToProps } from 'react-redux';

import { bindActionCreators } from 'redux';
import { acknowledgeNotification } from '../../actions';
import NotificationBadge, { DispatchProps } from '../../components/notification/NotificationBadge';

const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = (dispatch: Dispatch<any>) =>
  bindActionCreators(
    {
      handleAcknowledgeNotification: acknowledgeNotification
    },
    dispatch
  );

export default connect(
  null,
  mapDispatchToProps
)(NotificationBadge);
