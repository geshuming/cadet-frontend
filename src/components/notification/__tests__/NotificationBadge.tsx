import * as React from 'react';

import { mount } from 'enzyme';
import NotificationBadge from '../NotificationBadge';
import { Notification } from '../notificationShape';

const studentNotifications: Notification[] = [
  {
    id: 1,
    type: 'new',
    assessment_id: 1,
    assessment_type: 'Mission',
    assesssment_title: 'The Secret to Streams'
  },
  {
    id: 2,
    type: 'graded',
    assessment_id: 1,
    assessment_type: 'Mission',
    assesssment_title: 'The Secret to Streams'
  },
  {
    id: 3,
    type: 'autograded',
    assessment_id: 1,
    assessment_type: 'Mission',
    assesssment_title: 'The Secret to Streams'
  },
  {
    id: 4,
    type: 'unsubmitted',
    assessment_id: 1,
    assessment_type: 'Mission',
    assesssment_title: 'The Secret to Streams'
  }
];

const avengerNotifications: Notification[] = [
  {
    id: 1,
    type: 'submitted',
    assessment_id: 1,
    assessment_type: 'Mission',
    assesssment_title: 'The Secret to Streams',
    submission_id: 1
  }
];

const handleAcknowledgeNotification = (ids: number[]) => null;

describe('Badge', () => {
  test('renders properly for students', () => {
    const tree = mount(
      <NotificationBadge
        notifications={studentNotifications}
        handleAcknowledgeNotification={handleAcknowledgeNotification}
      />
    );

    expect(tree.debug()).toMatchSnapshot();
  });

  test('renders properly for avengers', () => {
    const tree = mount(
      <NotificationBadge
        notifications={avengerNotifications}
        handleAcknowledgeNotification={handleAcknowledgeNotification}
      />
    );

    expect(tree.debug()).toMatchSnapshot();
  });

  test('does not render with no notifications', () => {
    const tree = mount(
      <NotificationBadge
        notifications={[]}
        handleAcknowledgeNotification={handleAcknowledgeNotification}
      />
    );

    expect(tree.debug()).toMatchSnapshot();
  });
});
