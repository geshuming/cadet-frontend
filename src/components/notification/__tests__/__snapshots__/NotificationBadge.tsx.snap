// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge does not render with no notifications 1`] = `
"<NotificationBadge number={0}>
  <Blueprint3.Tag intent=\\"danger\\" round={true} large={[undefined]}>
    <span className=\\"bp3-tag bp3-intent-danger bp3-round\\" tabIndex={[undefined]}>
      <Blueprint3.Icon icon={[undefined]} />
      <Blueprint3.Text className=\\"bp3-fill\\" ellipsize={true} tagName=\\"span\\">
        <span className=\\"bp3-text-overflow-ellipsis bp3-fill\\" title={[undefined]}>
          0
        </span>
      </Blueprint3.Text>
      <Blueprint3.Icon icon={[undefined]} />
    </span>
  </Blueprint3.Tag>
</NotificationBadge>"
`;

exports[`Badge renders properly for avengers 1`] = `
"<NotificationBadge number={1}>
  <Blueprint3.Tag intent=\\"danger\\" round={true} large={[undefined]}>
    <span className=\\"bp3-tag bp3-intent-danger bp3-round\\" tabIndex={[undefined]}>
      <Blueprint3.Icon icon={[undefined]} />
      <Blueprint3.Text className=\\"bp3-fill\\" ellipsize={true} tagName=\\"span\\">
        <span className=\\"bp3-text-overflow-ellipsis bp3-fill\\" title={[undefined]}>
          1
        </span>
      </Blueprint3.Text>
      <Blueprint3.Icon icon={[undefined]} />
    </span>
  </Blueprint3.Tag>
</NotificationBadge>"
`;

exports[`Badge renders properly for students 1`] = `
"<NotificationBadge number={4}>
  <Blueprint3.Tag intent=\\"danger\\" round={true} large={[undefined]}>
    <span className=\\"bp3-tag bp3-intent-danger bp3-round\\" tabIndex={[undefined]}>
      <Blueprint3.Icon icon={[undefined]} />
      <Blueprint3.Text className=\\"bp3-fill\\" ellipsize={true} tagName=\\"span\\">
        <span className=\\"bp3-text-overflow-ellipsis bp3-fill\\" title={[undefined]}>
          4
        </span>
      </Blueprint3.Text>
      <Blueprint3.Icon icon={[undefined]} />
    </span>
  </Blueprint3.Tag>
</NotificationBadge>"
`;
